models/productDao.js

```diff
@@ -2,7 +2,14 @@ const mysqlDataSource = require('./dataSource');
 const QueryBuilder = require('./productListQuery');
 
 const getProducts = async (limit) => {
-  limit = Number(limit);
+  const queryBuilder = new QueryBuilder(
+    (mainCategoryId = undefined),
+    (subCategoryId = undefined),
+    (formulation = undefined),
+    (scent = undefined),
+    limit
+  );
```

Comment:
다음과 같이 리팩토링 해볼 수도 있습니다.

```suggestion
 const queryBuilder = new QueryBuilder({
  mainCategoryId: undefined,
  subCategoryId: undefined,
  formulation: undefined,
  scent: undefined,
  limit: limit,
});
```